tests/integration/test_upgrade.py

```diff
@@ -0,0 +1,257 @@
+@pytest.fixture(scope="module", name="gen_upgrade_test_charm_config")
+def gen_upgrade_test_charm_config_fixture(ops_test, swift_config, kube_core_client):
+    """Create a function that generates charm config for upgrading tests."""
+    swift_url = swift_config["swift-url"]
+    swift_config = copy.copy(swift_config)
+    swift_config["url"] = f"{swift_url}/{swift_config['bucket']}/wp-content/uploads/"
+    del swift_config["swift-url"]
+    swift_config["prefix"] = swift_config["object-prefix"]
+    del swift_config["object-prefix"]
+
+    def _gen_upgrade_test_charm_config():
+        charm_config = {
+            "db_host": kube_core_client.read_namespaced_pod(
+                name="mysql", namespace=ops_test.model_name
+            ).status.pod_ip,
+            "db_name": "wordpress",
+            "db_user": "wordpress",
+            "db_password": "wordpress-password",
+        }
+        if swift_config:
+            logger.info(f"deploy with swift: {swift_config}")
+            charm_config["wp_plugin_openstack-objectstorage_config"] = json.dumps(swift_config)
+        return charm_config
+
+    return _gen_upgrade_test_charm_config
+
+
+@pytest.mark.abort_on_fail
+async def test_deploy_old_version(
+    num_units,
+    ops_test: pytest_operator.plugin.OpsTest,
+    application_name,
+    kube_core_client,
+    deploy_and_wait_for_mysql_pod,
+    gen_upgrade_test_charm_config,
+):
+    """
+    arrange: none.
+    act: deploy all required charms and kubernetes pods for tests.
+    assert: all charms and pods are deployed successfully.
+    """
+    assert ops_test.model
+
+    async def deploy_wordpress():
+        await ops_test.model.deploy(
+            "wordpress-k8s",
+            resources={"wordpress-image": "wordpresscharmers/wordpress:v5.9.4-20.04_edge"},
+            application_name=application_name,
+            num_units=num_units,
+        )
+
+    await asyncio.gather(
+        deploy_wordpress(),
+        deploy_and_wait_for_mysql_pod(),
+        ops_test.run("playwright", "install", "chromium"),
+    )
+    await ops_test.model.applications[application_name].set_config(gen_upgrade_test_charm_config())
+    await ops_test.model.wait_for_idle(status=ops.model.ActiveStatus.name)  # type: ignore
+
+
+@pytest.mark.abort_on_fail
+async def test_create_example_blog(
+    ops_test,
+    default_admin_password,
+    unit_ip_list,
+    application_name,
+    test_image,
+    screenshot_dir,
+    kube_core_client,
+    swift_config,
+):
+    """
+    arrange: all charms and pods are deployed successfully.
+    act: enable some plugins and create posts and upload some images to the test WordPress blog.
+    assert: all operations finished without error.
+    """
+    namespace = ops_test.model_name
+
+    def get_wordpress_podspec_pod():
+        for pod in kube_core_client.list_namespaced_pod(namespace=namespace).items:
+            name = pod.metadata.name
+            if name.startswith("wordpress") and not name.endswith("-0"):
```

Comment:
Can't you use labels to find the pod instead of its name ?

Reply:
Fixed, thanks!